modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -81,13 +97,70 @@ export default function AuditBarChart({rtl, vocabularies}) {
 		? [BAR_CHART.dotRadiusMin]
 		: [BAR_CHART.legendHeight + BAR_CHART.dotRadiusMin];
 
+	const [checkboxes, setCheckbox] = useState(legendCheckboxes);
+
+	useEffect(() => {
+		if (Object.keys(colors).length) {
+			const head =
+				document.head || document.getElementsByTagName('head')[0];
+			const style = document.createElement('style');
+			style.id = 'legend-checkboxes';
+			style.type = 'text/css';
+
+			Object.entries(colors).forEach(([dataKey, color]) => {
+				style.textContent += `.custom-control-color-${dataKey}.custom-control-input:checked ~ 
+					.custom-control-label::before {
+						background-color: ${color};
+						border-color: ${color};
+				}
+				.custom-control-color-${dataKey}.custom-control-input:not(:checked) ~ 
+					.custom-control-label::before {
+						border-color: ${color};
+				}`;
+			});
```

Comment:
To avoid repeatedly accessing (reading then overwriting) the `.textContent` property, you could `reduce()` here too. Warning: this is almost certainly pointless microoptimization, but I like the way it puts the interesting information right up front (ie. before I even get to the `reduce()` I can see that the purpose of all this and the end result is going to be an assignment to `style.textContent`).

```js
if (Object.keys(colors).length) {
	const head = document.head || document.getElementsByTagName("head")[0];
	const style = document.createElement("style");
	style.id = "legend-checkboxes";

	style.type = "text/css";

	style.textContent = Object.entries(colors).reduce(
		(acc, [dataKey, color]) => {
			return acc.concat(`
				.custom-control-color-${dataKey}.custom-control-input:checked ~ 
					.custom-control-label::before {
						background-color: ${color};
						border-color: ${color};
				}
				.custom-control-color-${dataKey}.custom-control-input:not(:checked) ~ 
					.custom-control-label::before {
						border-color: ${color};
				}
			`);
		}
	);

	head.appendChild(style);
}
```

Reply:
I ❤️ easy reading